tests/functional/mesh/work_test.go

```diff
@@ -587,6 +607,140 @@ func TestWork(t *testing.T) {
+			err = controllers["node3"].AssertWorkPending(ctx, unitID)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// read the file size for the work
+			workStatus, err := controllers["node2"].GetWorkStatus(unitID)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// modify the mesh to break connections betweeen all the nodes
+			modifiedData := mesh.YamlData{}
+			modifiedData.Nodes = make(map[string]*mesh.YamlNode)
+
+			modifiedData.Nodes["node2"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{},
+				Nodedef: []interface{}{
+					map[interface{}]interface{}{
+						"tcp-listener": map[interface{}]interface{}{
+							"cost": 1.0,
+							"nodecost": map[interface{}]interface{}{
+								"node1": 1.0,
+								"node3": 1.0,
+							},
+						},
+					},
+				},
+			}
+			modifiedData.Nodes["node1"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{},
+				Nodedef:     []interface{}{},
+			}
+			modifiedData.Nodes["node3"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{},
+				Nodedef:     []interface{}{},
+			}
+
+			// modify the existing mesh , ( change `nodedef.yaml` for each node )
+			err = mesh.ModifyCLIMeshFromYaml(modifiedData, *m)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// reload the entire mesh
+			for node := range modifiedData.Nodes {
+				err = controllers[node].Reload()
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+
+			// modify the mesh to have connection again
+			withConnectionData := mesh.YamlData{}
+			withConnectionData.Nodes = make(map[string]*mesh.YamlNode)
+			withConnectionData.Nodes["node2"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{},
+				Nodedef: []interface{}{
+					map[interface{}]interface{}{
+						"tcp-listener": map[interface{}]interface{}{
+							"cost": 1.0,
+							"nodecost": map[interface{}]interface{}{
+								"node1": 1.0,
+								"node3": 1.0,
+							},
+						},
+					},
+				},
+			}
+			withConnectionData.Nodes["node1"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{
+					"node2": {
+						Index: 0,
+					},
+				},
+				Nodedef: []interface{}{},
+			}
+			withConnectionData.Nodes["node3"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{
+					"node2": {
+						Index: 0,
+					},
+				},
+				Nodedef: []interface{}{},
+			}
+
+			err = mesh.ModifyCLIMeshFromYaml(withConnectionData, *m)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// reload the entire mesh
+			// wait for 20 seconds
+			time.Sleep(20 * time.Second)
+			for node := range modifiedData.Nodes {
+				err = controllers[node].Reload()
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+
+			// wait for mesh to become ready again
+			ctx, _ = context.WithTimeout(context.Background(), 60*time.Second)
+			err = m.WaitForReady(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// so that the rsults can start streaming again
+			time.Sleep(10 * time.Second)
```

Comment:
I've rethought this. Instead of getting the first work results before the mesh is broken, let's get the work results after it's broken. Then when mesh re-establishes, we can check that stdout is larger.

Reply:
okay , i will make that change.